substratevm/src/com.oracle.svm.jfr/src/com/oracle/svm/jfr/JfrBuffer.java

```diff
@@ -56,6 +56,12 @@ static int offsetOfPos() {
         throw VMError.unimplemented(); // replaced
     }
 
+    @RawField
+    Pointer getTop();
```

Comment:
I think the top pointer is needed for the implementation in this PR because of the EventWriter view that is modified when flushing Java buffers to global buffer but isn't (currently) modified when writing Java buffers to disk at safepoint.

The buffers don't get reinitialized when writing to disk. If they did, we would also need to execute an update of the EventWriter similar to:
https://github.com/native-image-jfr/graal/blob/jfr/substratevm/src/com.oracle.svm.jfr/src/com/oracle/svm/jfr/SubstrateJVM.java#L340

I think it makes sense to handle it this way for the long term because
* The "lazy" reinitialize shouldn't increase cost for threads writing many events and will reduce cost for threads that write few events (less than size of thread local buffer)
* The work done in the uninterruptible safepoint operation should be as minimal as possible
* It uses the same infrastructure and pattern as concurrent writes of buffers to disk so less will need to be changed when making that optimization